app/models/similarity/unit_similarity_module.rb

```diff
@@ -222,4 +235,60 @@ def add_done_files_for_plagiarism_check_of(task_definition, tmp_path, to_check,
 
     self
   end
+
+  # JPLAG Function - extracts "done" files for each task and packages them into a directory for JPLAG to run on
+  def run_jplag_on_done_files(task_definition, tasks_dir, tasks_with_files, unit_code)
+    type_data = task_definition.moss_language.split
+    return if type_data.nil? || (type_data.length != 2) || (type_data[0] != 'moss')
+    similarity_pct = task_definition.plagiarism_warn_pct
+    return if similarity_pct.nil?
+
+    # Check if the directory exists and create it if it doesn't
+    results_dir = "/jplag/results/#{unit_code}"
+    `sudo docker exec jplag sh -c 'if [ ! -d "#{results_dir}" ]; then mkdir -p "#{results_dir}"; fi'`
+
+    # Remove existing result file if it exists
+    result_file = "#{results_dir}/#{task_definition.id}-result.zip"
+    `sudo docker exec jplag sh -c 'if [ -f "#{result_file}" ]; then rm "#{result_file}"; fi'`
+
+    # get each code file for each task
+    task_definition.upload_requirements.each_with_index do |upreq, idx|
+      # only check code files marked for similarity checks
+      next unless upreq['type'] == 'code' && upreq['tii_check']
+
+      pattern = task_definition.glob_for_upload_requirement(idx)
+
+      tasks_with_files.each do |t|
+        t.extract_file_from_done(tasks_dir, pattern, ->(_task, to_path, name) { File.join(to_path.to_s, t.student.username.to_s, name.to_s) })
+      end
+
+      logger.info "Starting JPLAG container to run on #{tasks_dir}"
+      root_dir = Rails.root.to_s
+      tasks_dir_split = tasks_dir.to_s.split(root_dir)[1]
+
+      # Set the file language based on the type
```

Comment:
We can be removing MOSS with this - so the languages can be updated to those we want from JPlag. This would be part of the front end change as well.